### PR TITLE
Fix rotation of images with odd number of rows

### LINKED ISSUE #6924: JPEG image renders with a horizontal line across middle of image when Exif Orientation tag changes
## Short description

When opening a JPEG image, depending on the Exif Orientation tag of the image, the RawTherapee UI may render a horizontal line across the middle of the image as in the following example:

rt-ui

When the image is saved from RawTherapee, the horizontal line is also saved to the resulting file.

## Steps to reproduce

1. Open the following image in RawTherapee: 
2. Observe the horizontal line in the middle of the image in the UI.

## Expected behavior

The UI does not render a horizontal line in the middle of the image.

### RawTherapee version
```
Version: 5.9
Branch: releases
Commit: 9b8583988
Commit date: 2022-11-27
Compiler: cc 12.2.0
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.24.7
Lensfun: V0.3.3.0
Build type: Release
Build flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -flto=auto -std=c++11 -ffp-contract=off  -Werror=unused-label -Werror=delete-incomplete -fno-math-errno -Wno-attributes -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -Wunused-macros -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
Link flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -flto=auto
OpenMP support: ON
MMAP support: ON
Build OS: 
Build date:  UTC
Build epoch: 
Build UUID: 
```

### OS version
Linux guava 6.6.8-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:01:01 +0000 x86_64 GNU/Linux

### Other information
When the Exif Orientation is changed (from 8 to 3), RawTherapee loads the image correctly.
rt-ui-ok

```
% exiv2 -pv 000154400034.jpg | grep Image.*Orientation
0x0112 Image        Orientation                 Short       1  8  # <- Problematic image has this orientation
0x0112 Image        Orientation                 Short       1  3  # <- Working as expected image has this orientation
```

### PR DESCRIPTION
**Pull request by Lawrence37**
For images with an odd number of rows, a 180 degree rotation resulted in one row of un-rotated pixels because it was effectively rotated twice.

Closes #6924.

### CODE REVIEW

File: rtengine/iimage.h [hunk 1]
```diff
@@ -415,6 +415,21 @@ class PlanarWhateverData : virtual public ImageDatas
...
+            // Middle row of odd-height images: only go half way otherwise the
+            // pixels will be swapped twice.
+            if (height & 1) {
+                int i = height / 2;
+                int width2 = width / 2;
+                for (int j = 0; j < width2; j++) {
+                    T tmp;
+                    int x = width - 1 - j;
+
+                    tmp = v(i, j);
```
Comment (Floessie) [inline rtengine/iimage.h]: @Lawrence37 Isn't that a prime example for [`std::swap()`](https://en.cppreference.com/w/cpp/algorithm/swap)? And if not, I'd rather have it `const T tmp = v(i, j);`...

Comment (Lawrence37) [reply rtengine/iimage.h]: I was being lazy (and cautious) and copy-pasted the code from above. I will see if swap is usable here.

Comment (Lawrence37) [reply rtengine/iimage.h]: Replaced with `std::swap`

**Comment by Lawrence37** [comment]
I will merge it this weekend if there are no objections.

**Pull request merged**